lib/srv/usermgmt_linux.go

```diff
@@ -71,3 +77,62 @@ func (*HostUsersProvisioningBackend) DeleteUser(name string) error {
 	}
 	return trace.Wrap(err)
 }
+
+// CheckSudoers ensures that a sudoers file to be written is valid
+func (*HostUsersProvisioningBackend) CheckSudoers(contents []byte) error {
+	err := host.CheckSudoers(contents)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func writeSudoersFile(root, name string, data []byte) (string, error) {
+	f, err := os.CreateTemp(root, fmt.Sprintf("tmp.%s.%s", "teleport", name))
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	defer f.Close()
+
+	if _, err = f.Write(data); err != nil {
+		return f.Name(), trace.Wrap(err)
+	}
+	if err = f.Chmod(0640); err != nil {
+		return f.Name(), trace.Wrap(err)
+	}
+	return f.Name(), nil
+}
+
+// WriteSudoersFile creates the user's sudoers file.
+func (u *HostUsersProvisioningBackend) WriteSudoersFile(username string, contents []byte) error {
+	if err := u.CheckSudoers(contents); err != nil {
+		return trace.Wrap(err)
+	}
+	sudoersFilePath := filepath.Join(u.sudoersPath, fmt.Sprintf("%s-%s-%s", "teleport", u.hostUUID, username))
+	if utils.FileExists(sudoersFilePath) {
+		return trace.Errorf("sudoers file %q already exists", sudoersFilePath)
+	}
```

Comment:
This check is racy and there's no real way (I think) to fix it, let's just document that different sudoers directives for the same host user will result in "unspecified" behavior - the latest file that happens to be renamed onto the file wins, rather than the "first" one as this check would suggest.

```suggestion
```

Reply:
(renaming a file onto an existing one works and atomically replaces the existing one)